test/unit/bot.spec.ts

```diff
@@ -0,0 +1,54 @@
+import { Bot } from '../../src/bots/bot'
+
+test('hello with correct name', () => {
+  expect('Hello Jon').toBe('Hello Jon')
+})
+
+// ------ tests to do ------
+// -- create bot
+test('create bot', () => {
+  const bot = new Bot('username', 'password', 100)
+  expect(bot).toBeDefined()
+})
+// -- login hit count
+test('create bot', () => {
+  const times = Math.ceil(Math.random() * 10)
+
+  const bot1 = new Bot('username', 'password', 100)
+  // simulating 100% login hit count
+  Array(times).fill(1).forEach(x => {
+    bot1.foundLogin(true)
+  })
+  expect(bot1.getLoginHitCount()).toBe(100)
+
+  const bot2 = new Bot('username', 'password', 100)
+  // simulating alternate login hit count
+  Array(times).fill(1).forEach((x, i) => {
+    if(i % 2 === 0) {
+      console.log('--calling foundLogin(true)')
+      bot2.foundLogin(true)
+    } else {
+      console.log('--calling foundLogin(false)')
+      bot2.foundLogin(false)
+    }
+  })
+  const loginFoundTimes = Array(times).fill(1).filter((x, i) => i % 2 === 0).length
+  expect(bot1.getLoginHitCount()).toBe(100 * loginFoundTimes / times)
+})
+
+// -- captcha hit count
+test('create bot', () => {
+  const bot = new Bot('username', 'password', 100)
+  const times = Math.ceil(Math.random() * 10)
+  Array(times).fill(1).forEach(x => {
+    bot.foundCaptcha(true)
+  })
+  expect(bot.getCaptchaHitCount()).toBe(100)
+})
+
+// -- rate limit
+// -- TODO
+test('create bot', () => {
+  const bot = new Bot('username', 'password', 100)
+  expect(bot).toBeDefined()
+})
```

Comment:
Not a good idea to call all of those tests `create bot`. One of them fails and it can't really tell which one. I think the second one.

The poor naming in addition with the randomness makes it very convoluted as well. Needs a big refactor. Best to keep randomness to few tests as possible. The one tests fails only sometimes - which is horrible for unit tests.

Reply:
Fixed names. I wanted to simulate calling the function a random number of times while randomly setting foundLogin or foundCaptcha.
Will see if i can determinatively hit them odd+even number of times.